### PR TITLE
formula: add support for generating click shell completions

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Add support for generating Python Click completions.

Ref: https://click.palletsprojects.com/en/8.1.x/shell-completion/#enabling-completion
> Click will output it when called with `_{PROG_NAME}_COMPLETE` set to `{shell}_source`. `{PROG_NAME}` is the executable name in uppercase with dashes replaced by underscores. The built-in shells are `bash`, `zsh`, and `fish`.

Note that Click doesn't support macOS Bash (needs 4.4+) and will fail if generation is attempted there.

-----

Particular formulae that could use this:
- [`hatch`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/hatch.rb#L231-L232)
- [`pipenv`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/pipenv.rb#L75-L78)
- [`ykman`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/ykman.rb#L97-L98)

---

`sqlite-utils` should also be able to use this and maybe some others.

I locally tried with `sqlite-utils` adding
```rb
generate_completions_from_executable(bin/"sqlite-utils", shells: [:fish, :zsh], shell_parameter_format: :click)
```

Snippets of generated files:
```console
❯ cat /opt/homebrew/opt/sqlite-utils/share/fish/vendor_completions.d/sqlite-utils.fish | head -5
function _sqlite_utils_completion;
    set -l response;

    for value in (env _SQLITE_UTILS_COMPLETE=fish_complete COMP_WORDS=(commandline -cp) COMP_CWORD=(commandline -t) sqlite-utils);
        set response $response $value;

❯ cat /opt/homebrew/opt/sqlite-utils/share/zsh/site-functions/_sqlite-utils | head -5
#compdef sqlite-utils

_sqlite_utils_completion() {
    local -a completions
    local -a completions_with_descriptions
```

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
> Particular formulae that could use this:

Do we have any formulae already generating these completions and/or users requesting this?

**Comment by cho-m** [comment]
> Do we have any formulae already generating these completions and/or users requesting this?

The 3 I listed are the ones already generating the completions.

File: Library/Homebrew/formula.rb [hunk 1]
```diff
@@ -1760,6 +1761,10 @@ def generate_completions_from_executable(*commands,
         "--shell=#{shell}"
       elsif shell_parameter_format == :none
         nil
+      elsif shell_parameter_format == :click
```
Comment (EricFromCanada) [inline Library/Homebrew/formula.rb:1764]: Should add this to the method's `@param shell_parameter_format` comment as well.

**Comment by cho-m** [comment]
Should be ready for final review and merge.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @cho-m!

**Pull request merged**